test/pylib/scylla_cluster.py

```diff
@@ -836,11 +837,11 @@ async def _before_test(self, test_case_name: str) -> str:
         self.logger.info("Setting up %s", self.current_test_case_full_name)
         if self.cluster.is_dirty:
             self.logger.info(f"Current cluster %s is dirty after last test, stopping...", self.cluster.name)
-            await self.clusters.steal()
             await self.cluster.stop()
             await self.cluster.release_ips()
             self.logger.info(f"Waiting for new cluster for test %s...", self.current_test_case_full_name)
-            await self._get_cluster()
```

Comment:
Wouldn't it be enough to reorder get_cluster and steal() here as I suggested in our phone call? Why?

Reply:
I recall the suggestion to reorder `stop()` and `steal()`, which wouldn't be enough, a new run might obtain a cluster between the `steal()` and `_get_cluster()`, allowing the concurrent run to start, as explained in the commit message.

I don't recall reordering `steal()` and `_get_cluster()` being suggested. In any case, it would lead to deadlocks, since existing runs would try to obtain new clusters before returning the currently held ones. Imagine pool size 4, 4 concurrent runs, they all call `_get_cluster()`. They would hang.